### ISSUE TITLE
advanced search throws error on spatial search

### ISSUE THREAD

**Issue by paregorios** [isawnyu/pleiades-gazetteer#218]
## steps to reproduce

As an anonymous user:
1. Visit http://pleiades.stoa.org/search_form
2. Select the "Geolocation Parameters" tab
3. Enter "27.445,5.322" in the "Lower Left Corner" box
4. Enter "96.024, 39.933" in the "Upper Right Corner" box
5. Enter "5.0" in the "Distance Tolerance" box
6. Select the "Workflow Details" tab
7. De-select all the checkboxes under "Item Type" and then select only the checkbox for "Place"
8. Select the "Advanced Search" button
9. Make a cup of coffee
10. View the results
11. Select the "Retrieve a KML representation using these search terms" button
12. Drink the cup of coffee
13. Observe the error message
## traceback

```
rLog 1472823922.130.80234649215 http://pleiades.stoa.org/search_kml
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module pleiades.kml.browser, line 176, in __call__
  Module Products.Five.browser.pagetemplatefile, line 125, in __call__
  Module Products.Five.browser.pagetemplatefile, line 59, in __call__
  Module zope.pagetemplate.pagetemplate, line 132, in pt_render
   - Warning: Macro expansion failed
   - Warning: <type 'exceptions.KeyError'>: 'main_macros'
  Module zope.pagetemplate.pagetemplate, line 240, in __call__
  Module collective.newrelic.patches.talinterpreter, line 17, in monkeypatch
  Module newrelic.api.function_trace, line 110, in literal_wrapper
  Module zope.tal.talinterpreter, line 271, in __call__
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 888, in do_useMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 946, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 821, in do_loop_tal
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 852, in do_condition
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 888, in do_useMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 851, in do_condition
  Module Products.PageTemplates.Expressions, line 205, in evaluateBoolean
  Module zope.tales.tales, line 696, in evaluate
   - URL: /srv/python27-apps/pleiades4/src/pleiades.kml/pleiades/kml/kml_macros.pt
   - Line 61, Column 6
   - Expression: <PathExpr standard:u'pm/hasMultiLineString'>
   - Names:
      {'args': (),
       'container': <PloneSite at /plone>,
       'context': <PloneSite at /plone>,
       'default': <object object at 0x7f172b470bb0>,
       'here': <PloneSite at /plone>,
       'loop': {},
       'nothing': None,
       'options': {},
       'repeat': <Products.PageTemplates.Expressions.SafeMapping object at 0x7f16ec38a2b8>,
       'request': <HTTPRequest, URL=http://pleiades.stoa.org/search_kml>,
       'root': <Application at >,
       'template': <Products.Five.browser.pagetemplatefile.ViewPageTemplateFile object at 0x7f16bb7e7590>,
       'traverse_subpath': [],
       'user': <SpecialUser 'Anonymous User'>,
       'view': <Products.Five.metaclass.PleiadesSearchDocument object at 0x7f16cfcf5710>,
       'views': <Products.Five.browser.pagetemplatefile.ViewMapper object at 0x7f16b871e350>}
  Module zope.tales.expressions, line 217, in __call__
  Module Products.PageTemplates.Expressions, line 147, in _eval
  Module zope.tales.expressions, line 124, in _eval
  Module Products.PageTemplates.Expressions, line 97, in trustedBoboAwareZopeTraverse
  Module zope.traversing.adapters, line 136, in traversePathElement
   - __traceback_info__: (<pleiades.kml.browser.AggregationPlacemark instance at 0x7f16cfe44d88>, 'hasMultiLineString')
  Module zope.traversing.adapters, line 50, in traverse
   - __traceback_info__: (<pleiades.kml.browser.AggregationPlacemark instance at 0x7f16cfe44d88>, 'hasMultiLineString', [])
LocationError: (<pleiades.kml.browser.AggregationPlacemark instance at 0x7f16cfe44d88>, 'hasMultiLineString')
```

**Comment by davisagli** [comment]
This was a bug in my fix for #201; I just deployed a fix to production.

**Comment by skleinfeldt** [comment]
You're supposed to be on vacation @davisagli !!
